test/basic/memory.wast

```diff
@@ -19,3 +19,24 @@
 )
 (assert_return (invoke "grow" (i32.const 0x1000)) (i32.const 0))
 (assert_return (invoke "grow" (i32.const 0x10000)) (i32.const -1))
+
+(module
+  (memory 1)
+  (func (export "loadInt64") (param $offset i32) (result i64)
+    (i64.load (local.get $offset))
+  )
+  (func (export "storeInt64") (param $offset i32) (param $value i64)
+    (i64.store (local.get $offset) (local.get $value))
+  )
+  (func (export "loadDouble") (param $offset i32) (result f64)
+    (f64.load (local.get $offset))
+  )
+  (func (export "storeDouble") (param $offset i32) (param $value f64)
+    (f64.store (local.get $offset) (local.get $value))
+  )
+)
+;; memory is 4 kilobyte aligned, therefore index 1 will not be four byte aligned.
+(assert_return (invoke "storeInt64" (i32.const 0x1) (i64.const 0x0123456789abcdef)))
```

Comment:
There should be a loadInt8 check as well, which checks that the last byte is really written on offset 1. On earlier ARMs, the lower bits of the offsets were ignored, so this test could pass even if the code does not work.